ext/stackprof/extconf.rb

```diff
@@ -1,5 +1,7 @@
 require 'mkmf'
-if have_func('rb_postponed_job_register_one') &&
+if RUBY_ENGINE == 'truffleruby'
+  fail "try truffleruby's profiler \nruby --experimental-options --cpusampler --cpusampler.Mode=roots --cpusampler.SampleInternal FILE.rb"
```

Comment:
Not being a `TruffleRuby` user myself, I assume that the reasoning for this PR is that `stackprof` is not support on that platform (JVM?).  Maybe include that as part of the string here:

```ruby
fail "stackprof not supported on this platform (#{RUBY_ENGINE})!\n" \
     "for truffleruby, consider trying out truffleruby's profiler:\n\n" \
     "  $ ruby --experimental-options --cpusampler --cpusampler.Mode=roots --cpusampler.SampleInternal FILE.rb"
```

Reply:
Also, not sure the `extconf.rb` is the best place for this, though I don't have an alternative off hand to suggest.

Reply:
Maybe `mkmf` can be a better place but it is a more complex thing.